fusedrug/data/interaction/drug_target/datasets/dti_binding_dataset.py

```diff
@@ -120,21 +117,37 @@ def dti_binding_dataset(
             verify_integrity=True,
         )
 
+    if pairs_columns_to_extract is not None:
+        pairs_df = pairs_df[pairs_columns_to_extract]
+    if ligands_columns_to_extract is not None:
+        ligands_df = ligands_df[ligands_columns_to_extract]
+    if targets_columns_to_extract is not None:
+        targets_df = targets_df[targets_columns_to_extract]
+    if pairs_rename_columns is not None:
+        pairs_df = pairs_df.rename(pairs_rename_columns, axis=1)
+    if ligands_rename_columns is not None:
+        ligands_df = ligands_df.rename(ligands_rename_columns, axis=1)
+    if targets_rename_columns is not None:
+        targets_df = targets_df.rename(targets_rename_columns, axis=1)
+
+    if return_dataframes:
+        return {"pairs": pairs_df, "ligands": ligands_df, "targets": targets_df}
+
     dynamic_pipeline = [
         (
             OpReadDataframe(
                 pairs_df,
-                columns_to_extract=pairs_columns_to_extract,
-                rename_columns=pairs_rename_columns,
+                columns_to_extract=None,
+                rename_columns=None,
```

Comment:
not sure about the `columns_to_extract` part,
it's possibly better to still keep using it in the op to make sure it's not too "heavy" per sample.
@mosheraboh do you think it matters or not ?

I think that you wanted mainly the rename to happen (also) in the dataframe.

Any option that you and @alex-golts agree on is fine by me.

Reply:
Was it (pairs_columns_to_extract ) set to None before? If not, I suggest keeping it.